File: class4.1/CustomPizza.cpp
#include "CustomPizza.h"

void CustomPizza::addTopping(const std::string& topping, double cost) {
    toppings.push_back(topping);
    totalCost += cost;
}

void CustomPizza::display() const {
    std::cout << "Toppings:" << std::endl;
    for (const auto& topping : toppings) {
        std::cout << "- " << topping << std::endl;
    }
    std::cout << "Total cost: $" << totalCost << std::endl;
}

double CustomPizza::getTotalCost() const {
    return totalCost;
}

File: home4.1/CustomPizza.java
import java.util.ArrayList;
import java.util.List;

public class CustomPizza {
    private List<String> toppings;
    private double totalCost;

    public CustomPizza() {
        this.toppings = new ArrayList<>();
        this.totalCost = 0.0;
    }

    public void addTopping(String topping, double cost) {
        toppings.add(topping);
        totalCost += cost;
    }

    public void display() {
        System.out.println("Toppings:");
        for (String topping : toppings) {
            System.out.println("- " + topping);
        }
        System.out.println("Total cost: $" + totalCost);
    }

    public double getTotalCost() {
        return totalCost;
    }
}
